File: controllers/authController.py
from fastapi import Depends, HTTPException
from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm
from services.authService import authenticate_user

oauth2_scheme = OAuth2PasswordBearer(tokenUrl='token')

async def login(form_data: OAuth2PasswordRequestForm = Depends()):
    user = authenticate_user(form_data.username, form_data.password)
    if not user:
        raise HTTPException(status_code=400, detail='Incorrect username or password')
    return {'access_token': user.username, 'token_type': 'bearer'}

File: controllers/authController.js
const authService = require('../services/authService');
const { User } = require('../models/User');

exports.login = async (req, res) => {
	try {
		const { email, password } = req.body;
		const token = await authService.login(email, password);
		res.status(200).json({ token });
	} catch (error) {
		res.status(400).json({ message: error.message });
	}
};

exports.register = async (req, res) => {
	try {
		const user = await authService.register(req.body);
		res.status(201).json(user);
	} catch (error) {
		res.status(400).json({ message: error.message });
	}
};